src/sql/parts/objectExplorer/common/objectExplorerService.ts

```diff
@@ -320,41 +356,95 @@ export class ObjectExplorerService implements IObjectExplorerService {
 				let newRequest = false;
 				if (!self._sessions[session.sessionId].nodes[nodePath]) {
 					self._sessions[session.sessionId].nodes[nodePath] = {
-						expandEmitter: new Emitter<sqlops.ObjectExplorerExpandInfo>()
+						expandEmitter: new Emitter<NodeExpandInfoWithProviderId>()
 					};
 					newRequest = true;
 				}
-				self._sessions[session.sessionId].nodes[nodePath].expandEmitter.event(((expandResult) => {
-					if (expandResult && !expandResult.errorMessage) {
-						resolve(expandResult);
-					}
-					else {
-						reject(expandResult ? expandResult.errorMessage : undefined);
+				let provider = this._providers[providerId];
+				if (provider) {
+					let resultMap: Map<string, sqlops.ObjectExplorerExpandInfo> = new Map<string, sqlops.ObjectExplorerExpandInfo>();
+					let allProviders: sqlops.ObjectExplorerProviderBase[] = [provider];
+
+					let nodeProviders = this._nodeProviders[providerId];
+					if (nodeProviders) {
+						nodeProviders = nodeProviders.sort((a, b) => a.group.toLowerCase().localeCompare(b.group.toLowerCase()));
+						allProviders.push(...nodeProviders);
 					}
+
+					self._sessions[session.sessionId].nodes[nodePath].expandEmitter.event((expandResult) => {
+						if (expandResult) {
+							if (expandResult.providerId) {
+								resultMap.set(expandResult.providerId, expandResult);
+							} else {
+								reject(expandResult.errorMessage ? expandResult.errorMessage : 'ProviderId is undefined in expandResult');
+							}
+						}
+						else {
+							reject(expandResult ? expandResult.errorMessage : undefined);
+						}
+
+						// When get all responses from all providers, merge results
+						if (resultMap.size === allProviders.length) {
+							resolve(self.mergeResults(allProviders, resultMap));
+
+							// Have to delete it after get all reponses otherwise couldn't find session for not the first response
+							if (newRequest) {
+								delete self._sessions[session.sessionId].nodes[nodePath];
+							}
+						}
+					});
 					if (newRequest) {
-						delete self._sessions[session.sessionId].nodes[nodePath];
+						allProviders.forEach(provider => {
+							TelemetryUtils.addTelemetry(this._telemetryService, TelemetryKeys.ObjectExplorerExpand, { refresh: 0, provider: providerId });
```

Comment:
Please review this with Matt. I think we should only send this event once, with primary providerId (e.g. `mssql`) listed. Sending 1 for each provider doesn't make sense to me. As this is a possible usage metric let's be careful about this and assume we should only send once.